core/src/test/java/com/zextras/carbonio/files/api/GetPublicNodeApiIT.java

```diff
@@ -242,4 +242,204 @@ void givenAPublicLinkIdAndAnExistingFileTheGetPublicNodeShouldReturnThePublicFil
         .hasSize(1)
         .containsExactly("Could not find link with id abcd1234abcd1234abcd1234abcd1234");
   }
+
+  @Test
+  void givenAPublicLinkIdWithAccessCodeAndAnExistingFolderTheGetPublicNodeWithCorrectCodeShouldReturnThePublicFolder() {
+    // Given
+    long now = System.currentTimeMillis();
+    DatabasePopulator.aNodePopulator(simulator.getInjector())
+        .addNode(
+            new PopulatorNode(
+                "00000000-0000-0000-0000-000000000000",
+                "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
+                "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
+                "LOCAL_ROOT",
+                "folder",
+                "",
+                NodeType.FOLDER,
+                "LOCAL_ROOT",
+                0L,
+                null))
+        .addLink(
+            "8cac6df0-3ecb-451d-a953-10c3ac5e3ebc",
+            "00000000-0000-0000-0000-000000000000",
+            "abcd1234abcd1234abcd1234abcd1234",
+            Optional.empty(),
+            Optional.empty(),
+            Optional.of("fake-access-code"));
+
+    String bodyPayload =
+        GraphqlCommandBuilder.aQueryBuilder("getPublicNode")
+            .withString("node_link_id", "abcd1234abcd1234abcd1234abcd1234")
+            .withString("access_code", "fake-access-code")
+            .withWantedResultFormat("{ id created_at updated_at name type }")
+            .build();
+
+    final HttpRequest httpRequest = HttpRequest.of("POST", "/public/graphql/", null, bodyPayload);
+
+    // When
+    HttpResponse httpResponse = TestUtils.sendRequest(httpRequest, simulator.getNettyChannel());
+
+    // Then
+    Assertions.assertThat(httpResponse.getStatus()).isEqualTo(200);
+
+    final Map<String, Object> publicNode =
+        TestUtils.jsonResponseToMap(httpResponse.getBodyPayload(), "getPublicNode");
+
+    Assertions.assertThat(publicNode.get("id")).isEqualTo("00000000-0000-0000-0000-000000000000");
+    Assertions.assertThat((long) publicNode.get("created_at")).isGreaterThanOrEqualTo(now);
+    Assertions.assertThat((long) publicNode.get("updated_at")).isGreaterThanOrEqualTo(now);
+    Assertions.assertThat(publicNode.get("name")).isEqualTo("folder");
+    Assertions.assertThat(publicNode.get("type")).isEqualTo(NodeType.FOLDER.toString());
+  }
+
+  // This tests that a node that is not a folder gets returned by getPublicNode even if access code is not passed
+  // for its public link
+  @Test
+  void givenAPublicLinkIdWithAccessCodeAndAnExistingFileTheGetPublicNodeWithoutAccessCodeShouldReturnThePublicFile() {
```

Comment:
Lol, this is so wrong! Mark my word: in the future this will be seen as a security bug :rofl: